# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/remoteworkers/RemoteWorkerService.h
# Commit: b0cfee4dccb7
# Full Hash: b0cfee4dccb70dbb0fd958d7430e1d0ce11fbdd8
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2025-03-18 21:32:34
# Regressor Bug: 1899503
# File Overlap Count: 4
# Description:
#   Bug 1899503 - P3 PRemoteWorkerDebugger binding implementation. r=asuth
#   
#   This patch implements the PRemoteWorkerDebugger binding.
#   
#   We are trying to keep the same logic debugger registration/unregistration to minimize the difference between the local debugger(the debugger on the content process main thread)
# ==============================================================================

diff -r 8973242af52a -r b0cfee4dccb7 dom/workers/remoteworkers/RemoteWorkerService.h
--- a/dom/workers/remoteworkers/RemoteWorkerService.h	Tue Mar 18 12:10:53 2025 +0000
+++ b/dom/workers/remoteworkers/RemoteWorkerService.h	Tue Mar 18 12:10:54 2025 +0000
@@ -19,6 +19,7 @@
 namespace mozilla::dom {
 
 class RemoteWorkerDebuggerManagerChild;
+class RemoteWorkerDebuggerManagerParent;
 class RemoteWorkerService;
 class RemoteWorkerServiceChild;
 class RemoteWorkerServiceShutdownBlocker;
@@ -123,7 +124,8 @@
 
   nsCOMPtr<nsIThread> mThread;
   RefPtr<RemoteWorkerServiceChild> mActor;
-  RefPtr<RemoteWorkerDebuggerManagerChild> mDebuggerManagerActor;
+  RefPtr<RemoteWorkerDebuggerManagerChild> mDebuggerManagerChild;
+  RefPtr<RemoteWorkerDebuggerManagerParent> mDebuggerManagerParent;
   // The keep-alive is set and cleared on the main thread but we will hand out
   // additional references to it from the "Worker Launcher" thread, so it's
   // appropriate to use a mutex.  (Alternately we could have used a ThreadBound